code/target/src/main/java/com/adobe/marketing/mobile/target/TargetPreviewManager.java

```diff
@@ -362,20 +380,32 @@ private boolean setupTargetPreviewParameters(final Map<String, String> previewPa
      *
      */
     private void createAndShowFloatingButton() {
-        if (floatingButton != null) {
+        if (floatingButtonPresentable != null) {
             Log.debug(TargetConstants.LOG_TAG, CLASS_NAME, "createAndShowFloatingButton - Floating button already exists");
             return;
         }
 
-        TargetPreviewButtonListener targetPreviewButtonListener = new TargetPreviewButtonListener(this);
-        floatingButton = uiService.createFloatingButton(targetPreviewButtonListener);
+        if (context.getAssets() == null) {
+            Log.debug(TargetConstants.LOG_TAG, CLASS_NAME, "createAndShowFloatingButton - Unable to instantiate the floating button for target preview, cannot find button image");
+            return;
+        }
 
-        if (floatingButton != null) {
-            floatingButton.display();
-        } else {
+        try {
+            final String backgroundImageString = StreamUtils.readAsString(context.getAssets().open("encoded_button_image.txt"));
```

Comment:
nit: Let's save the file name in TargetConstants and use, instead of hardcoding here.